Lab 1

Erbidium opened the pull request:
Remove readme


Add map for image


Add other pages of website


Remove other pages


Fix styles and replace div on header and footer


Lorem ipsum


Fixes and remove unused code


Add main instead of div


Remove unnecessary elements


Fix duplicating numbers in the list


Rework project structure and add different pictures


Rework project structure


Move lab1 to folder


Lab1 work


Fix husky


Edit .editorconfig


Add file for github actions

Erbidium committed:
Lab 1


Remove readme


Add map for image


Add other pages of website


Remove other pages


Fix styles and replace div on header and footer


Lorem ipsum


Fixes and remove unused code


Add main instead of div


Remove unnecessary elements


Fix duplicating numbers in the list


Rework project structure and add different pictures


Rework project structure


Move lab1 to folder


Lab1 work


Fix husky


Edit .editorconfig


Add file for github actions


Add stylelint dependencies


Reformat css file


Fix eslint script


Add prettier devdependencies


Reformat json files


Edit script eslint

nosovk commented on .idea/inspectionProfiles/Project_Default.xml:
```diff
@@ -0,0 +1,7 @@
+<component name="InspectionProjectProfileManager">
```
вероятно это тоже можно не коммитить

nosovk commented on .prettierignore:
```diff
@@ -0,0 +1 @@
+*.css
```
prettier как раз и нужен для всего не js, то есть для css в т.ч.

nosovk commented on lab1/sources/1.html:
```diff
@@ -0,0 +1,96 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="«utf-8»" />
```
тут ошибка

nosovk commented on lab1/sources/1.html:
```diff
@@ -0,0 +1,96 @@
+  <head>
+    <meta charset="«utf-8»" />
+    <title>Start page</title>
+    <link rel="stylesheet" href="styles.css" />
```
пропущен viewport

nosovk commented on lab1/sources/1.html:
```diff
@@ -0,0 +1,96 @@
+    <link rel="stylesheet" href="styles.css" />
+  </head>
+  <body>
+    <div class="mainBlock">
```
в css не принят camelcase

nosovk commented on lab1/sources/1.html:
```diff
@@ -0,0 +1,96 @@
+    <div class="mainBlock">
+      <header>
+        <div class="textHeaderWrapper">
+          <h1 id="headText">TESLA MODEL S</h1>
```
id is evil, avoid them

nosovk commented on lab1/sources/1.html:
```diff
@@ -0,0 +1,96 @@
+        </div>
+        <nav>
+          <div><a href="index.html">Home</a></div>
+          <div><a href="2.html">Page 2</a></div>
```
aboslute links

nosovk commented on lab1/sources/styles.css:
```diff
@@ -0,0 +1,165 @@
+nav a {
+  text-decoration: none;
+  font-size: 2em;
+  color: aliceblue;
```
css vars?

nosovk commented on package.json:
```diff
@@ -0,0 +1,42 @@
+  "main": "index.js",
+  "scripts": {
+    "prepare": "husky install",
+    "lint": "eslint **/*.js --no-error-on-unmatched-pattern",
```
use gitignore for ignore pattern instead of file

nosovk commented on .github/workflows/github-actions.yml:
```diff
@@ -0,0 +1,8 @@
+  build_and_preview:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
```
add cache for npm pls

Erbidium committed:
Lab1 fix

Fix prettierignore, errors in html files, add viewport, remove id, camelCase from css, fix gitignore instead of patterns in scripts, Fix gitignore and remove idea folder
Fix absolute links
Remove camelCase
Add css vars
Fix
Format is prettier
Fix
Fix
Add cache for npm
Edit .yml
